hapi-utils/src/main/java/com/hedera/services/sysfiles/domain/throttling/ThrottleBucket.java

```diff
@@ -213,6 +213,8 @@ private long lcm(long a, long b) {
 	}
 
 	private long gcd(long a, long b) {
+		// The assert is to comfort sonar. assertMinimalOpsPerSec already guarantees this, but sonar is not smart enough.
+		assert(a != 0 || b != 0);
```

Comment:
I don't think we need to comfort sonar at all cost. If it is not smart enough there, it's sonar's problem. And there should be an issue filed on their side!

Reply:
I agree we don't need to comfort sonar at all cost. This we need a little more thinking.

Reply:
I think that unless -ea is passed to the JVM it will skip the assertion during runtime. Is that right?
Does Sonar have a way to disable an error on a single line of code?

I think the answer to those questions will help us to understand whether this line is really worth it.

Reply:
By default. The assert is disabled. By doing this, sonar seems not complaining.

Yeah, sonar has couple of ways to disable reporting errors. I think this should be a better solution in this case.

My original thought was to not resort to sonar's feature to disable reporting errors if it's not a big effort.